sktime/forecasting/base/convertIO/_convertIO.py

```diff
@@ -0,0 +1,190 @@
+# -*- coding: utf-8 -*-
+"""
+Machine type converters for scitypes
+
+Exports
+-------
+convert : dictionary of functions, with signature as below
+        indexed by triples of types
+         1st element = convert from - type
+         2nd element = convert to - type
+         3rd element = considered as this scitype - string
+        elements are conversion functions of machine type (1st) -> 2nd
+
+convert_to : function/dispatch version of convert, with 1st elt inferred
+
+
+-----------------------------------------
+
+Function signature of
+convert[(from_type, to_type, as_scitype)]
+
+Parameters
+----------
+what : from_type - object to convert
+store : reference of storage for lossy conversions, default=None (no store)
+
+Returns
+-------
+converted_what : to_type - object what converted to to_type
+
+Raises
+------
+ValueError and TypeError, if requested conversion is not possible
+                            (depending on conversion logic)
+
+-----------------------------------------
+
+Function signature of convert_to
+
+Parameters
+----------
+what : object to convert (any type)
+to_type : type, the type to convert "what" to
+as_scitype : str - name of scitype the object "what" is considered as
+store : reference of storage for lossy conversions, default=None (no store)
+
+Returns
+-------
+converted_what : to_type - object what converted to to_type
+
+Raises
+------
+TypeError if no suitable key in convert exists
+    i.e., (type(what), to_type, as_scitype)
+
+
+copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""
+
+
+__author__ = ["fkiraly"]
+
+
+import numpy as np
+import pandas as pd
+
+
+# dictionary indexed by triples of types
+#  1st element = convert from - type
+#  2nd element = convert to - type
+#  3rd element = considered as this scitype - string
+# elements are conversion functions of machine type (1st) -> 2nd
+
+convert = dict()
+
+
+def convert_identity(what, store=None):
+
+    return what
+
+
+# assign identity function to type conversion to self
+for tp in [pd.Series, pd.DataFrame, np.array]:
+    convert[(tp, tp, "Series")] = convert_identity
+
+
+def convert_UvS_to_MvS_as_Series(what: pd.Series, store=None) -> pd.DataFrame:
+
+    if not isinstance(what, pd.Series):
+        raise TypeError("input must be a pd.Series")
```

Comment:
I wonder why you wrote `input must be a pd.Series` instead of `what must be a pd.Series`...

😛

Reply:
what do you mean?

Reply:
The error message you raise is `input must be a pd.Series`. I was wondering (in a tongue-in-cheek way), why you did not use the error message `what must be be a pd.Series`, given that the variable name is `what`.

Less cryptically, I am implying that `what must be a pd.Series` is a confusing error message, and thus, `what` is a confusing variable name.

Reply:
oh, I see - what would be a better name than what?

Reply:
I made a couple of suggestions in a comment above, https://github.com/alan-turing-institute/sktime/pull/980#discussion_r659212321. Copied here:

Some alternatives:
* `object` (used in python docs for generic inputs, e.g. in isinstance)
* `data` (because I think we will always be using this function to convert data, as opposed to other kinds of things)

To help think of ideas, is there a standard name for the kind of thing that y, X, Z etc. are?

Reply:
it's `obj` now, that's the arg name that appears in other python type related things like `isinstance`